### ISSUE TITLE
404 Error on multipage articles

### ISSUE THREAD

**Issue by scorp13** [joomla/joomla-cms#18042]
### Steps to reproduce the issue
Create multipage article with content like this:

```html
some intro text
<hr id="system-readmore" />
some 1st page text
<hr class="system-pagebreak" />
some 2nd page text
<hr class="system-pagebreak" />
some 3rd page text
```

and try to access page 2 or 3

**The main point is presence of `<hr id="system-readmore" />` before `<hr class="system-pagebreak" />`**
### Actual result
404 error when trying to access a page other than the first.

### System information (as much as possible)
Joomla 3.8

### Additional comments
The reason in the code added by the [commit](https://github.com/joomla/joomla-cms/commit/605a2f1f6d8445cab030209404f9050430bc4572#diff-a2fbd25d7bac224516450abdd42cd57b)
Code:
```php
		// Check if the intro text needs to be processed as well
		if ($item->introtext && strpos($item->text, $item->introtext) !== 0)
		{
			// Save the old text of the article
			$text = $item->text;

			// Set the intro text as new text
			$item->text = $item->introtext;

			// Trigger the event with the introtext
			$dispatcher->trigger('onContentPrepare', array ('com_content.article', &$item, &$item->params, $offset));

			// Set the prepared intro text back
			$item->introtext = $item->text;

			// Restore the original text variable
			$item->text = $text;
		}
```

In the case of the article's structure I described (which is quite common), after

```php
			// Set the intro text as new text
			$item->text = $item->introtext;
```

$item->text does not contain code `<hr class="system-pagebreak" />`

Next, in the onContentPrepare event handler of PageBreak plugin there is an error when trying to access the 2nd or more page:

```php
// Simple performance check to determine whether bot should process further.
if (StringHelper::strpos($row->text, 'class="system-pagebreak') === false)
{
	if ($page > 0)
	{
		throw new Exception(JText::_('JERROR_PAGE_NOT_FOUND'), 404);
	}
	
	return true;
}
```

**Comment by sandewt** [comment]
It seems a known problem !!!
See the issues: #17830 and #18016 .<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18042">issues.joomla.org/tracker/joomla-cms/18042</a>.</sub>

**Comment by ghost** [comment]
closed as duplicate Report ##17830

Thanks for Hint @sandewt<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18042">issues.joomla.org/tracker/joomla-cms/18042</a>.</sub>
